orttraining/orttraining/core/graph/gradient_builder_base.h

```diff
@@ -100,6 +102,15 @@ class GradientBuilderBase {
     return ArgDef(node_->InputDefs()[i]->Name(), node_->InputDefs()[i]->TypeAsProto());
   }
 
+  int IC(const size_t i) const {
+    auto input_name = I(i).name;
```

Comment:
The interface of this function should be 

bool IsStashedForBackward(const ArgDef& arg_def); 

1. we could also be checking if an O(0) has been used... 
2. return bool to more intuitive usage

Reply:
Why do we need ArgDef? Name should be sufficient, no?